enhancements/cluster-logging/loki-storage.md

```diff
@@ -0,0 +1,628 @@
+
+* Deploy Loki **only** in microservices mode, i.e. each component is deployed and managed as an individual deployment/statefulset. This enables individual capacity planning and resource utilization per path ingestion/querying.
+
+* Multi-tenancy is enabled per default and requires the collector to apply the required [tenant identification](https://grafana.com/docs/loki/latest/operations/multi-tenancy/) on ingestion time. No user can disable this setting. In turn queries require the same identification information to be successful. Furthermore there is no support for querying across tenants.
+
+* Loki is operated only in a zero-dependency setup mode as described [here](https://grafana.com/docs/loki/latest/configuration/examples/#almost-zero-dependencies-setup). The only externally managed requirement is the availability of an S3-compatible API for object storage to persist logs.
+
+* Support a limited set of Loki scale outs, which are battle-tested on our internally managed Loki deployment. The target scale outs are defined using the format `Nx.{extra-small, small, medium}` where `N` the number of Loki deployments:
+  * **1x.extra-small**: Dedicated scale for development and demo purposes only! Not a valid configuration for cluster logging production environments.
+  * **1x.small**: Dedicated scale for low-retention log storage without redundancy support. All components are deployed with HA by running at least two replicas.
+  * **1x.medium**: Dedicated scale for log-retention log storage with single redundancy support. All components with HA by running at least two replicas except the ingester requiring at minimum three replicas to allow failover.
+
+* Support tuning of CPU/Memory resource and limits for each individual component beyond the selected scale-out size.
+
+* Support adding kubernetes tolerations and node selectors for all Loki components individually. This enables operating the Loki deployment on dedicated nodes.
+
+* Support a limited set of control options from Loki's [limit_config](https://grafana.com/docs/loki/latest/configuration/#limits_config) for the ingestion and query paths. The set is an opinionated list of options to enable tuning of object storage usage. The list of options is offered globally across the whole Loki deployment or per tenant.
+
+For **1x.medium** instances only:
+
+* Support optionally single redundancy for log store on the target object storage.
+
+* Support auto-compaction of the index and chunks stored on the object storage when single redundancy enabled.
+
+### User Stories
+
+* As a cluster logging admin, I want to be able to configure Loki as an alternative log store in the ClusterLogging custom resource.
+
+* As a cluster logging admin, I want to be able to split logs to Loki as an alterantive log store in the ClusterLogForwarder custom resources to be able to support custom tenants.
```

Comment:
What does "split logs to Loki" mean here? Forward as in Cluster Log Forwarding?

Reply:
Split Logs refers here to `ClusterLogForwarder`'s capabilities to split log by type, i.e. `app`, `infra`, `audit`.